test/ex_machina/ecto_test.exs

```diff
@@ -44,18 +45,38 @@ defmodule ExMachina.EctoTest do
     }
   end
 
-  test "params_for/2 raises when passed a map" do
-    assert_raise ArgumentError, fn ->
-      TestFactory.params_for(:user_map)
-    end
-  end
-
   test "params_for/2 removes fields with nil values" do
     assert TestFactory.params_for(:user, admin: nil) == %{
-      name: "John Doe"
+      name: "John Doe",
+      articles: [],
     }
   end
 
+  test "params_for/2 recursively strips has_many values" do
+    article = TestFactory.build(:article)
+
+    user_params = TestFactory.params_for(:user, articles: [article])
+
+    assert user_params[:articles] == [%{
+      title: article.title,
+    }]
+  end
+
+  test "params_for/2 recursively strips has_one values" do
+    article = TestFactory.build(:article)
+
+    user_params = TestFactory.params_for(:user, best_article: article)
+
+    assert user_params[:best_article] == %{title: article.title}
+  end
+
+  test "params_for/2 works with maps as has_many values" do
+    article = %{title: "Foobar"}
+    user_params = TestFactory.params_for(:user, articles: [article])
```

Comment:
Oh you are right. It seems like there should be a newline in between there. I didn't check the code closely enough. Good comment!